### PR TITLE
fix(react-query): wrap queryOptions to support UseInfiniteQueryOptions

### PR DESCRIPTION
**Pull request by gwansikk**
# Overview

- #1015 

I have temporarily wrapped queryOptions to correspond to UseInfiniteQueryOptions. https://github.com/toss/suspensive/issues/1015#issuecomment-2199263821

## PR Checklist

- [x] I did below actions if need

1. I read the [Contributing Guide](https://github.com/toss/suspensive/blob/main/CONTRIBUTING.md)
2. I added documents and tests.

### CODE REVIEW

File: packages/react-query-4/src/useSuspenseInfiniteQuery.ts [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
+  type Query,
   type QueryKey,
   type UseInfiniteQueryOptions,
   type UseInfiniteQueryResult,
   useInfiniteQuery,
 } from '@tanstack/react-query'
 import type { OmitKeyof } from './utility-types'
 
+/**
+ * @experimental This is experimental feature.
```
Comment (gwansikk) [inline packages/react-query-4/src/useSuspenseInfiniteQuery.ts:12]: I will label it as `experimental` until `infiniteQueryOptions` is introduced.

File: packages/react-query-4/src/useSuspenseInfiniteQuery.ts [hunk 2]
```diff
@@ -23,12 +32,23 @@ export interface UseSuspenseInfiniteQueryOptions<
...
+  > {
+  refetchOnMount?: RefetchOption<TQueryFnData, TError, TData, TQueryKey>
+  refetchOnReconnect?: RefetchOption<TQueryFnData, TError, TData, TQueryKey>
+  refetchOnWindowFocus?: RefetchOption<TQueryFnData, TError, TData, TQueryKey>
+}
 
 /**
  * This hook is wrapping useInfiniteQuery of `@tanstack/react-query` v4 with default suspense option.
  * @see {@link https://suspensive.org/docs/react-query/useSuspenseInfiniteQuery}
+ * @experimental This is experimental feature.
```
Comment (gwansikk) [inline packages/react-query-4/src/useSuspenseInfiniteQuery.ts:51]: I will label it as `experimental` until `infiniteQueryOptions` is introduced.

**Comment by manudeli** [comment]
@gwansikk

1. coauthor @kangju2000 please
2. add changset please

File: packages/react-query-4/src/SuspenseInfiniteQuery.test-d.tsx [hunk 3]
```diff
@@ -60,6 +61,20 @@ describe('<SuspenseInfiniteQuery/>', () => {
...
+    ;() => (
+      <SuspenseInfiniteQuery {...infiniteQueryOptions}>
+        {(query) => {
+          expectTypeOf(query).toEqualTypeOf<UseSuspenseInfiniteQueryResult<{ text: string }>>()
+          expectTypeOf(query.data).toEqualTypeOf<InfiniteData<{ text: string }>>()
+          expectTypeOf(query.status).toEqualTypeOf<'success'>()
+          return <></>
+        }}
+      </SuspenseInfiniteQuery>
+    )
```
Comment (manudeli) [inline packages/react-query-4/src/SuspenseInfiniteQuery.test-d.tsx:77]: select type test should be added

**Comment by gwansikk** [comment]
> @gwansikk
> 
> 1. coauthor @kangju2000 please
> 2. add changset please

1. Oh, I forgot! I'll add it!
2. Can I now decide and add the `changeset` myself?
3. Wrapping the select seems to be taking longer than expected... I'll do it tomorrow (07.02)!

**Comment by manudeli** [comment]
> Can I now decide and add the changeset myself?

Sure! you're already maintainer. You have access

image

**Comment by gwansikk** [comment]
@manudeli 
I attempted to wrap queryOptions to correspond to UseInfiniteQueryOptions. However, I found that the method I tried is complicated and difficult for users to use, especially with wrapping the `select` props, which is very complex. Therefore, I believe the best approach is to quickly create and provide `InfiniteQueryOptions`.

I will close this issue and focus on quickly creating and providing `InfiniteQueryOptions`.

**Comment by manudeli** [comment]
> @manudeli I attempted to wrap queryOptions to correspond to UseInfiniteQueryOptions. However, I found that the method I tried is complicated and difficult for users to use, especially with wrapping the `select` props, which is very complex. Therefore, I believe the best approach is to quickly create and provide `InfiniteQueryOptions`.
> 
> I will close this issue and focus on quickly creating and providing `InfiniteQueryOptions`.

I agree 🚀 Let's try it

**Pull request closed without merging**